src/iosrv/srv.c

```diff
@@ -1199,6 +1199,22 @@ dss_collective_reduce_internal(struct dss_coll_ops *ops,
 		stream			= &stream_args->csa_streams[tid];
 		stream->st_coll_args	= &carg;
 
+		if (args->ca_exclude_tgts_cnt) {
+			int i;
+
+			for (i = 0; i < args->ca_exclude_tgts_cnt; i++)
+				if (args->ca_exclude_tgts[i] == tid)
+					break;
+
+			if (i < args->ca_exclude_tgts_cnt) {
+				D_DEBUG(DB_TRACE, "Skip tgt %d\n", tid);
+				rc = ABT_future_set(future,
+					    (void *)&aggregator);
```

Comment:
hmm, although it works but seems better to be  "rc = ABT_future_set(future, (void *)stream);"

The original code of below L1229 seems also better to be changed as that.

Reply:
hmm, you mean collective_reduce should use stream instead of aggregator? or I misunderstood? I do not want to to touch that in this patch.

Reply:
I think it is fine that collective_reduce use the aggregator, above L1195 set the first future argement as aggregator.
But for specific xstream it uses stream argument, see the handling in collective_reduce is assumes the arg[0] is aggregator and others are dss_stream_arg_type.

The current code should work but just that not exactly how the arguments used. but not a big deal

Reply:
ah, right. thanks.